src/components/BookForm.js

```diff
@@ -0,0 +1,253 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { confirmAlert } from "react-confirm-alert";
+
+class BookForm extends React.Component {
+  static propTypes = {
+    bookFormContent: PropTypes.shape({
+      title: PropTypes.string,
+      author: PropTypes.string,
+      isbn: PropTypes.string,
+      rating: PropTypes.number,
+      length: PropTypes.number,
+      series: PropTypes.string,
+      textSnippet: PropTypes.string,
+      source: PropTypes.string,
+      note: PropTypes.string,
+      purchased: PropTypes.bool,
+      prioritize: PropTypes.bool
+    }).isRequired,
+    createNewBook: PropTypes.func.isRequired,
+    editBook: PropTypes.func.isRequired,
+    editing: PropTypes.bool.isRequired,
+    isbnList: PropTypes.array.isRequired,
+    saveBook: PropTypes.func.isRequired,
+    showEditForm: PropTypes.bool.isRequired,
+    toggleModal: PropTypes.func.isRequired
+  };
+
+  handleConfirmSubmit = e => {
+    e.preventDefault();
+    confirmAlert({
+      customUI: ({ onClose }) => {
+        return (
+          <div className="confirm-dialog">
+            <h3 className="confirm-dialog__title">Edit Existing Book?</h3>
+            <p className="confirm-dialog__message">
+              You are about to update a book that’s already in the book list.
+              Are you sure you want to do that? Pressing Cancel will return you
+              to the edit form. Double-check your ISBN number to avoid problems.
+            </p>
+            <div className="confirm-dialog__actions">
+              <button
+                className="btn btn-primary"
+                onClick={() => {
+                  this.props.saveBook();
+                  onClose();
+                }}
+              >
+                Edit Book
+              </button>
+              <button className="btn btn-secondary" onClick={onClose}>
+                Cancel
+              </button>
+            </div>
+          </div>
+        );
+      }
+    });
```

Comment:
Could this be a functional component? With a prop of like `text={You are about to update a book...}`

Reply:
Or I recall there being something about `children` in React components too. That could look nice.

Reply:
I'll be honest, I had a hell of a time getting this to work. It wants to live as a `confirmAlert` in a function. I have no idea how to go about breaking this into a separate component. For now, I'm going to leave this as-is, but holler at me if you think I'm really doing something dumb.